scripts/prepare_docker_buildinfo.sh

```diff
@@ -22,7 +22,45 @@ if [ -z "$DISTRO" ]; then
     [ -z "$DISTRO" ] && DISTRO=jessie
 fi
 
-DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo 
+version_control(){
+    version_file=files/build/versions/default/versions-docker
+    new_version_file=target/versions/build/versions-docker
+
+    dir=`dirname $new_version_file`
+    mkdir -p $dir
```

Comment:
How about simply change to "mkdir -p target/versions/build", since the variable dir is no used in the other places.
Or 
version_build_dir=target/versions/build
new_version_file=$version_build_dir/versions-docker
mkdir -p $version_build_dir